server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/create/CreateSnapshotResponse.java

```diff
@@ -93,4 +99,40 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         builder.endObject();
         return builder;
     }
+
+    public static CreateSnapshotResponse fromXContent(XContentParser parser) throws IOException {
+        CreateSnapshotResponse createSnapshotResponse = new CreateSnapshotResponse();
+
+        parser.nextToken(); // '{'
```

Comment:
This seems like we should be using a nested parser here. Similar to how this works, https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/action/admin/cluster/repositories/verify/VerifyRepositoryResponse.java#L112-L116

Reply:
This may be possible in a follow up PR, but I don't want to modify more existing content parsers and the current structure does not allow for one since the end array/object tokens do not align with the way the current object parser works.

Reply:
After reviewing the structure of the content here, I do not believe it makes sense to add a new object parser in this case since it would end up being more obfuscated for no reason since with one of the fields it's an either or and the field doesn't need to actually be parsed (it's used for user-consumable output only).  This would mean additional empty methods.  It's better to call fromXContent directly on SnapshotInfo.  I did, however, add several checks to ensure the proper tokens are being read.